twisted/test/proto_helpers.py

```diff
@@ -695,3 +702,86 @@ def connectUNIX(self, address, factory, timeout=30, checkPID=0):
         Fake L{IReactorUNIX.connectUNIX}, that raises L{_connectException}.
         """
         raise self._connectException
+
+
+
+@implementer(ILogFilterPredicate)
+class NamespaceFilterPredicate(object):
+    """
+    L{ILogFilterPredicate} that filters out events not matching configured
+    namespace.
+    """
+
+    def __init__(self, namespace):
+        self.namespace = namespace
+
+
+    def __call__(self, event):
+        namespace = event.get("log_namespace", None)
+
+        if namespace != self.namespace:
+            return PredicateResult.no
+
+        return PredicateResult.maybe
+
+
+
+class LogCapture(object):
```

Comment:
I don't think this belongs in the `proto_helpers` namespace.  It's generally useful to application code.  If you're concerned about exposing it to application developers prematurely, this is the wrong place to put it - `proto_helpers` is explicitly covered by the compatibility policy.  (Which was probably a mistake, we should eventually move everything to a better-named package, like `twisted.testing`.)  But most importantly, the `proto` in `proto_helpers` refers to protocols, and this has nothing to do with protocols.  So I leave it to you whether you want to make it public or not, but in any case it belongs in `twisted.logger`.